fish-rust/src/common.rs

```diff
@@ -2105,7 +2220,8 @@ mod tests {
 
     /// Test wide/narrow conversion by creating random strings and verifying that the original
     /// string comes back through double conversion.
-    pub fn test_convert() {
+    #[test]
+    fn test_convert() {
```

Comment:
Just for the sake of documentation: I experienced spurious fails of `test_convert` that did not fail when extracted into separate tests, one such example was

```rust
thread 'common::tests::test_convert' panicked at 'assertion failed: `(left == right)`
  left: `[117, 90, 37, 255, 59, 42, 157, 142, 207, 110, 38, 147, 106, 72, 208, 134, 203, 106, 4, 210, 175, 46, 129, 69, 160, 13, 127, 91, 176, 207, 187, 250, 140, 95, 248, 127, 38, 129, 21, 92, 123, 206, 248, 62, 171, 216, 101, 246, 238, 52, 43, 221, 74, 56, 100, 131, 214, 100, 177, 149, 181, 180, 56, 144, 207, 242, 177, 244, 180, 139, 228, 74, 178, 204, 25, 174, 8, 206, 76, 181, 13, 13, 115, 99, 194, 74, 64, 208, 51, 137, 164, 10, 163, 179, 199, 89, 252, 110, 198, 247, 24, 250, 120, 46, 237, 133, 208, 173, 109, 175, 196, 188, 239, 196, 107, 251, 184, 191, 136, 23, 174, 35, 200, 54, 124, 234, 209, 24, 49, 107, 121, 147, 228, 29, 28, 56, 223, 66, 65, 72, 49, 248, 202, 204, 167, 19, 238, 109, 40, 203, 105, 220, 194, 252, 128, 11, 123, 112, 90, 128, 72, 91, 2, 195, 160, 73, 150, 117, 207, 102, 219, 85, 178, 158, 166, 85, 13, 76, 178, 132]`,
 right: `[117, 90, 37, 255, 59, 42, 194, 157, 194, 142, 195, 143, 110, 38, 194, 147, 106, 72, 195, 144, 194, 134, 195, 139, 106, 4, 210, 175, 46, 129, 69, 160, 13, 127, 91, 176, 207, 187, 250, 140, 95, 248, 127, 38, 129, 21, 92, 123, 206, 248, 62, 171, 216, 101, 246, 238, 52, 43, 221, 74, 56, 100, 131, 214, 100, 177, 149, 181, 180, 56, 144, 207, 242, 177, 244, 180, 139, 228, 74, 178, 204, 25, 174, 8, 206, 76, 181, 13, 13, 115, 99, 194, 74, 64, 208, 51, 137, 164, 10, 163, 179, 199, 89, 252, 110, 198, 247, 24, 250, 120, 46, 237, 133, 208, 173, 109, 175, 196, 188, 239, 196, 107, 251, 184, 191, 136, 23, 174, 35, 200, 54, 124, 234, 209, 24, 49, 107, 121, 147, 228, 29, 28, 56, 223, 66, 65, 72, 49, 248, 202, 204, 167, 19, 238, 109, 40, 203, 105, 220, 194, 252, 128, 11, 123, 112, 90, 128, 72, 91, 2, 195, 160, 73, 150, 117, 207, 102, 219, 85, 178, 158, 166, 85, 13, 76, 178, 132]`: Conversion cycle of string:
 180 chars: 0x75 0x5A 0x25 0xFF 0x3B 0x2A 0x9D 0x8E 0xCF 0x6E 0x26 0x93 0x6A 0x48 0xD0 0x86 0xCB 0x6A 0x 4 0xD2 0xAF 0x2E 0x81 0x45 0xA0 0x D 0x7F 0x5B 0xB0 0xCF 0xBB 0xFA 0x8C 0x5F 0xF8 0x7F 0x26 0x81 0x15 0x5C 0x7B 0xCE 0xF8 0x3E 0xAB 0xD8 0x65 0xF6 0xEE 0x34 0x2B 0xDD 0x4A 0x38 0x64 0x83 0xD6 0x64 0xB1 0x95 0xB5 0xB4 0x38 0x90 0xCF 0xF2 0xB1 0xF4 0xB4 0x8B 0xE4 0x4A 0xB2 0xCC 0x19 0xAE 0x 8 0xCE 0x4C 0xB5 0x D 0x D 0x73 0x63 0xC2 0x4A 0x40 0xD0 0x33 0x89 0xA4 0x A 0xA3 0xB3 0xC7 0x59 0xFC 0x6E 0xC6 0xF7 0x18 0xFA 0x78 0x2E 0xED 0x85 0xD0 0xAD 0x6D 0xAF 0xC4 0xBC 0xEF 0xC4 0x6B 0xFB 0xB8 0xBF 0x88 0x17 0xAE 0x23 0xC8 0x36 0x7C 0xEA 0xD1 0x18 0x31 0x6B 0x79 0x93 0xE4 0x1D 0x1C 0x38 0xDF 0x42 0x41 0x48 0x31 0xF8 0xCA 0xCC 0xA7 0x13 0xEE 0x6D 0x28 0xCB 0x69 0xDC 0xC2 0xFC 0x80 0x B 0x7B 0x70 0x5A 0x80 0x48 0x5B 0x 2 0xC3 0xA0 0x49 0x96 0x75 0xCF 0x66 0xDB 0x55 0xB2 0x9E 0xA6 0x55 0x D 0x4C 0xB2 0x84 
produced different string:
 187 chars: 0x75 0x5A 0x25 0xFF 0x3B 0x2A 0xC2 0x9D 0xC2 0x8E 0xC3 0x8F 0x6E 0x26 0xC2 0x93 0x6A 0x48 0xC3 0x90 0xC2 0x86 0xC3 0x8B 0x6A 0x 4 0xD2 0xAF 0x2E 0x81 0x45 0xA0 0x D 0x7F 0x5B 0xB0 0xCF 0xBB 0xFA 0x8C 0x5F 0xF8 0x7F 0x26 0x81 0x15 0x5C 0x7B 0xCE 0xF8 0x3E 0xAB 0xD8 0x65 0xF6 0xEE 0x34 0x2B 0xDD 0x4A 0x38 0x64 0x83 0xD6 0x64 0xB1 0x95 0xB5 0xB4 0x38 0x90 0xCF 0xF2 0xB1 0xF4 0xB4 0x8B 0xE4 0x4A 0xB2 0xCC 0x19 0xAE 0x 8 0xCE 0x4C 0xB5 0x D 0x D 0x73 0x63 0xC2 0x4A 0x40 0xD0 0x33 0x89 0xA4 0x A 0xA3 0xB3 0xC7 0x59 0xFC 0x6E 0xC6 0xF7 0x18 0xFA 0x78 0x2E 0xED 0x85 0xD0 0xAD 0x6D 0xAF 0xC4 0xBC 0xEF 0xC4 0x6B 0xFB 0xB8 0xBF 0x88 0x17 0xAE 0x23 0xC8 0x36 0x7C 0xEA 0xD1 0x18 0x31 0x6B 0x79 0x93 0xE4 0x1D 0x1C 0x38 0xDF 0x42 0x41 0x48 0x31 0xF8 0xCA 0xCC 0xA7 0x13 0xEE 0x6D 0x28 0xCB 0x69 0xDC 0xC2 0xFC 0x80 0x B 0x7B 0x70 0x5A 0x80 0x48 0x5B 0x 2 0xC3 0xA0 0x49 0x96 0x75 0xCF 0x66 0xDB 0x55 0xB2 0x9E 0xA6 0x55 0x D 0x4C 0xB2 0x84 ', src/common.rs:2292:13
```

It did not fail if I made the test locale-sensitive. I am guessing it's related to high-memory pressure, as it no longer reproduces after a restart.